src/stands/StandEventHandler.cpp

```diff
@@ -201,6 +303,53 @@ namespace UKControllerPlugin {
                 message.at("callsign").is_string();
         }
 
+        /*
+            Get the airfield to use for stand assignment purposes, depending on how far the aircraft is from
+            its origin.
+        */
+        std::string StandEventHandler::GetAirfieldForStandAssignment(EuroScopeCFlightPlanInterface& flightplan)
+        {
+            return flightplan.GetDistanceFromOrigin() < this->maxDistanceFromDepartureAirport
+                ? flightplan.GetOrigin()
+                : flightplan.GetDestination();
+        }
+
+        /*
+            Every time the flightplan updates, keep the annotation up to date if it's
+            not so. This is needed because there wont be any flightplans to annotate
+            when the plugin first starts up, so it annotates them as they come along.
+        */
+        void StandEventHandler::FlightPlanEvent(
+            EuroScopeCFlightPlanInterface& flightPlan,
+            EuroScopeCRadarTargetInterface& radarTarget
+        ) {
+            if (!this->standAssignments.count(flightPlan.GetCallsign())) {
+                return;
+            }
+
+            if (
+                this->stands.find(this->standAssignments.at(flightPlan.GetCallsign()))->identifier ==
+                    flightPlan.GetAnnotation(this->annotationIndex)
+            ) {
+                return;
+            }
+
+            flightPlan.AnnotateFlightStrip(
+                this->annotationIndex,
+                this->stands.find(this->standAssignments.at(flightPlan.GetCallsign()))->identifier
+            );
+        }
+
+        void StandEventHandler::FlightPlanDisconnectEvent(EuroScopeCFlightPlanInterface& flightPlan)

```

Comment:
Can these just be defined as `default` in the header file?

Reply:
Apparently not - you can only do default for constructors and destructors

Reply:
how annoying - cheers